### ISSUE TITLE
3 compile problems on Gentoo Linux

### ISSUE THREAD

**Issue by onkelbeh** [markusressel/xs1-api-client#5]
Hi Markus,

thanks a lot for working on XS1 Gateway Component for Home Assitant. I do not use the component myself, but I am maintaining [Home Assistant Gentoo Overlay](https://github.com/onkelbeh/HomeAssistantRepository), and I am trying to keep it complete, so I want to find a way to compile all integrations Home Assistant uses with Gentoo's own package manager `portage`.

**Describe the bug**

First I hope you don't mind that I use your integration in Gentoo Overlay for Home Assistant. You can find the corresponding Ebuild files for it in https://github.com/onkelbeh/HomeAssistantRepository/tree/master/dev-python/xs1-api-client. During the last Python 3.7 compile test, I found some issues in compiling your source, which I just wanted to let you know:

1) Normally I would use the release file from Pypi in SDIST format, which you already provide, because this is technically the best source for Gentoo Python Ebuilds, it is able to use Gentoo's mirror system. Unfortunately these files are incomplete in all tested releases, 'requirements.txt' is missing in the archives, so setup.py fails. I had to change `SRC_URI` to github archives. Please note that (on Pypi) the name of the archive is not matching the package name:
```bash
# SDIST packackage at Pypi is missing 'requirements.txt'
#MY_PN=${PN//-/_}
#MY_P=${MY_PN}-${PV}
#SRC_URI="mirror://pypi/${P:0:1}/${MY_PN}/${MY_P}.tar.gz -> ${P}.tar.gz"
#S=${WORKDIR}/${MY_PN}-${PV}
SRC_URI="https://github.com/markusressel/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```
2) setup.py tries to get the current branch name from git, which does not work if you compile from an archive. In this particular case, there is no git, this should be catched.
```python
GIT_BRANCH = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"])
GIT_BRANCH = GIT_BRANCH.decode()  # convert to standard string
GIT_BRANCH = GIT_BRANCH.rstrip()  # remove unnecessary whitespace
```
I solved it (by now) with a dirty hack:
```bash
sed -i "s/subprocess.check_output(\[\"git\", \"rev-parse\", \"--abbrev-ref\", \"HEAD\"\])/'master'/" -i setup.py || die
sed -i "s/GIT_BRANCH = GIT_BRANCH.decode()  //" -i setup.py || die
```
3) setup.py tries to install a 'tests' package at top level, this is bad style at all, and forbidden in Gentoo. I had to change:
`packages=find_packages()`
to
`packages=find_packages(exclude=['tests','tests.*']`

`sed -i "s/packages=find_packages()/packages=find_packages(exclude=['tests','tests.*'])/g" -i setup.py || die`

fixed in: https://git.edevau.net/onkelbeh/HomeAssistantRepository/commit/f419a3406772105f959275ab720f434ed9fddecd

**Expected behavior**

already explained above

**Screenshots**

**Desktop (please complete the following information):**
 - OS: current Gentoo Linux on amd64, x86_64-pc-linux-gnu, also good for other platforms

**Additional context**

Perhaps we can find a way to make this a bit easier :-)

**Comment by markusressel** [comment]
Hi @onkelbeh,
I don't mind you supporting this integration.

As you can see [here](https://github.com/markusressel/xs1-api-client/blob/master/.github/workflows/python_publish.yaml) I use the setup.py and twine to upload new releases.

If I understand it correctly, the "git" issue is irrelevant if we can get the SDIST release to work correctly, right?
I have opened a PR with the proposed changes I would make, feel free to commend on that, but now that I see the failing build, I would like to switch from `requirements.txt` to `Pipfile.lock`, as I have done with many other projects of mine. Does that work for you too?

**Comment by onkelbeh** [comment]
Hi Markus,

thanks for your fast reponse.

Sorry, but I don't know anything about github workflows, all my scripts run local, I use github.com only as a mirror/backup.

#6 looks good to me, think this should add the missing file, did not test your exclude for find_packages yet, I think this won't hurt, but #6 would **not** replace/fix the missing git.

What about changing setup.py to:
```python
try:
  GIT_BRANCH = subprocess.check_output(["git", "rev-parse", "--abbrev-ref", "HEAD"])
  GIT_BRANCH = GIT_BRANCH.decode()  # convert to standard string
  GIT_BRANCH = GIT_BRANCH.rstrip()  # remove unnecessary whitespace
except:
  GIT_BRANCH = 'master'
```
I would throw it out, this code won't find anything else than a 'master' branch on releases anyway, and still you have to set the version number by hand. You never released an alpha or beta.

I'm keeping over 750 different Home Assistant components and nearly 1000 libraries in this repository, but I've never noticed code like that in a `setup.py` :-)

**Comment by markusressel** [comment]
@onkelbeh I have just released [v3.0.1](https://github.com/markusressel/xs1-api-client/releases/tag/v3.0.1), let me know if there is still something to work on.
